activerecord/test/cases/relations_test.rb

```diff
@@ -188,6 +189,30 @@ def test_finding_with_reverted_assoc_order
     assert_equal topics(:fifth).title, topics.first.title
   end
 
+  def test_reverse_order_with_function
+    topics = Topic.order("length(title)").reverse_order
+    assert_equal topics(:second).title, topics.first.title
+  end
+
+  def test_reverse_order_with_function_other_predicates
+    topics = Topic.order("author_name, length(title), id").reverse_order
+    assert_equal topics(:second).title, topics.first.title
+    topics = Topic.order("length(author_name), id, length(title)").reverse_order
+    assert_equal topics(:fifth).title, topics.first.title
+  end
+
+  def test_reverse_order_with_multiargument_function
+    assert_raises(ActiveRecord::IrreversibleOrderError) do
+      Topic.order("concat(author_name, title)").reverse_order
```

Comment:
The following in stable rails: `Topic.order("concat(author_name, title)").reverse_order.to_sql` will give you `order by concat(author_name desc, title) desc`.

The goal of this patch is not to improve a way how rails reverse order but clearly mark cases that rails can not handle.